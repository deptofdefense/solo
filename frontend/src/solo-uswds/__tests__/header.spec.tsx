import React from "react";
import { render, fireEvent, wait } from "test-utils";
import {
  Header,
  HeaderLogo,
  HeaderNav,
  HeaderNavLink,
  HeaderNavbar
} from "solo-uswds";

describe("uswds Header component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("uswds HeaderNavbar component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<HeaderNavbar />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("uswds HeaderLogo component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<HeaderLogo text="test" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("includes link when text is passed", () => {
    const { getByAltText } = render(<HeaderLogo text="test" />);
    expect(getByAltText("test")).toBeInTheDocument();
  });

  it("excludes link when text is not passed", () => {
    const { queryByAltText } = render(<HeaderLogo />);
    expect(queryByAltText("test")).not.toBeInTheDocument();
  });
});

describe("uswds HeaderNavComponent", () => {
  let props = {
    isOpen: true,
    onClose: jest.fn()
  };

  afterEach(() => {
    props.onClose.mockReset();
  });

  it("matches snapshot when open", () => {
    const { asFragment } = render(<HeaderNav {...props} isOpen />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot when closed", () => {
    const { asFragment } = render(<HeaderNav {...props} isOpen={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("adds visible class when open", () => {
    const { container } = render(<HeaderNav {...props} isOpen />);
    expect(container.firstChild).toHaveClass("is-visible");
  });

  it("omits visible class when closed", () => {
    const { container } = render(<HeaderNav {...props} isOpen={false} />);
    expect(container.firstChild).not.toHaveClass("is-visible");
  });

  it("calls onClose when close button is clicked", async () => {
    const { getByAltText } = render(<HeaderNav {...props} />);
    const closeIcon = getByAltText("close");
    fireEvent.click(closeIcon);
    await wait(() => {
      expect(props.onClose).toHaveBeenCalledTimes(1);
    });
  });
});

describe("uswds HeaderNavLink component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<HeaderNavLink to="#">test</HeaderNavLink>);
    expect(asFragment()).toMatchSnapshot();
  });
});
