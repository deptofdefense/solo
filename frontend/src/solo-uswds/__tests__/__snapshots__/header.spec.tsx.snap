// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uswds Header component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="usa-header usa-header--basic"
  >
    <div
      class="usa-nav-container navContainer"
    />
  </header>
</DocumentFragment>
`;

exports[`uswds HeaderLogo component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="usa-logo logoHeader"
  >
    <em
      class="usa-logo__text"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <img
          alt="test"
          class="logoImg"
          src="solo.png"
        />
      </a>
    </em>
  </div>
</DocumentFragment>
`;

exports[`uswds HeaderNavComponent matches snapshot when closed 1`] = `
<DocumentFragment>
  <nav
    aria-label="Primary navigation"
    class="usa-nav"
  >
    <ul
      class="usa-nav__primary usa-accordion"
    >
      <button
        class="usa-nav__close"
      >
        <img
          alt="close"
          src="close.svg"
        />
      </button>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`uswds HeaderNavComponent matches snapshot when open 1`] = `
<DocumentFragment>
  <nav
    aria-label="Primary navigation"
    class="usa-nav is-visible"
  >
    <ul
      class="usa-nav__primary usa-accordion"
    >
      <button
        class="usa-nav__close"
      >
        <img
          alt="close"
          src="close.svg"
        />
      </button>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`uswds HeaderNavLink component matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="usa-nav__primary-item"
  >
    <a
      aria-current="page"
      class="usa-nav__link usa-current"
      href="/"
    >
      <span>
        test
      </span>
    </a>
  </li>
</DocumentFragment>
`;

exports[`uswds HeaderNavbar component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="usa-navbar"
  />
</DocumentFragment>
`;
